src/main/java/marquez/core/services/JobService.java

```diff
@@ -0,0 +1,224 @@
+package marquez.core.services;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import marquez.core.exceptions.UnexpectedException;
+import marquez.core.models.Job;
+import marquez.core.models.JobRun;
+import marquez.core.models.JobRunState;
+import marquez.core.models.JobVersion;
+import marquez.core.models.RunArgs;
+import marquez.dao.JobDAO;
+import marquez.dao.JobRunDAO;
+import marquez.dao.JobVersionDAO;
+import marquez.dao.RunArgsDAO;
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
+
+@Slf4j
+class JobService {
+  private JobDAO jobDAO;
+  private JobVersionDAO jobVersionDAO;
+  private JobRunDAO jobRunDAO;
+  private RunArgsDAO runArgsDAO;
+
+  public JobService(
+      JobDAO jobDAO, JobVersionDAO jobVersionDAO, JobRunDAO jobRunDAO, RunArgsDAO runArgsDAO) {
+    this.jobDAO = jobDAO;
+    this.jobVersionDAO = jobVersionDAO;
+    this.jobRunDAO = jobRunDAO;
+    this.runArgsDAO = runArgsDAO;
+  }
+
+  public Optional<Job> getJob(String namespace, String jobName) throws UnexpectedException {
+    try {
+      return Optional.ofNullable(jobDAO.findByName(namespace, jobName));
+    } catch (UnableToExecuteStatementException e) {
+      String err = "failed to get a job";
+      log.error(err, e);
+      throw new UnexpectedException();
+    }
+  }
+
+  public Job createJob(String namespace, Job job) throws UnexpectedException {
+    try {
+      Job existingJob = this.jobDAO.findByName(namespace, job.getName());
```

Comment:
We should make sure there is testing to verify the behavior when there is no match (i.e. return null instead of throwing an exception).

Reply:
I'm not following -- it does not throw an exception if there's no match. It returns an optional with no value present.

In fact, there is such a test!
https://github.com/MarquezProject/marquez/pull/154/files#diff-c67c728ae0d22570bebb6e68e1266d7aR96

Reply:
Awesome! Reason I ask is b/c I've had the mappers throw an exception when there was no match, so I just want to make sure it's handled gracefully.